### PR TITLE
fix: make tsm-report JSON-friendly

### PR DESCRIPTION
**Pull request by thomas-fossati**

### CODE REVIEW

File: docs/tsm-report.cddl [hunk 1]
```diff
@@ -1,8 +1,15 @@
 tsm-report = {
-  ? "auxblob": bstr
-  "outblob": bstr
-  "provider": bstr
+  ? auxblob: binary-string
+  outblob: binary-string
+  provider: binary-string
```
Comment (deeglaze) [inline docs/tsm-report.cddl:4]:
```suggestion
  provider: tstr
```

File: docs/tsm-report.cddl [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
   ? service-report
 }
-service-report //= ("manifestblob": bstr, "service_provider": bstr)
-service-report //= ("service_provider": bstr)
+
+service-report = ((
+  manifestblob: binary-string
+  service_provider: binary-string
+) // service_provider: binary-string)
```
Comment (deeglaze) [inline docs/tsm-report.cddl:11]:
```suggestion
) // service_provider: tstr)
```

File: docs/tsm-report.cddl [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+  outblob: binary-string
+  provider: binary-string
   ? service-report
 }
-service-report //= ("manifestblob": bstr, "service_provider": bstr)
-service-report //= ("service_provider": bstr)
+
+service-report = ((
+  manifestblob: binary-string
+  service_provider: binary-string
```
Comment (deeglaze) [inline docs/tsm-report.cddl:10]:
```suggestion
  service_provider: tstr
```

**Review by deeglaze — APPROVED** [review]
The provider fields are human-readable, so let’s make them readable for json

File: docs/tsm-report.cddl [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
+  outblob: binary-string
+  provider: tstr
   ? service-report
 }
-service-report //= ("manifestblob": bstr, "service_provider": bstr)
-service-report //= ("service_provider": bstr)
+
+service-report = ((
+  manifestblob: binary-string
+  service_provider: tstr
```
Comment (jraman567) [inline docs/tsm-report.cddl:10]: Hi. service_provider is a WO attribute. So, how can we get the service_provider from TSM?

https://www.kernel.org/doc/Documentation/ABI/testing/configfs-tsm

**Comment by jraman567** [comment]
Thanks for this PR, @thomas-fossati . It looks good, except for the service_provider comment I made above.

Thanks, @deeglaze , for reviewing this also.

File: docs/tsm-report.cddl (hunk 4 above)
Comment (deeglaze) [reply docs/tsm-report.cddl:10]: Yikes, well, that can be included as part of the attestation scheme I guess. The manifest being write only seems like a bug on the kernel side though..

Comment (deeglaze) [reply docs/tsm-report.cddl:10]: Perhaps we should hold off on registering this type before we can test it

Comment (thomas-fossati) [reply docs/tsm-report.cddl:10]: @deeglaze could you test it?

**Comment by thomas-fossati** [comment]
> Thanks for this PR, @thomas-fossati . It looks good, except for the service_provider comment I made above.

Thanks, @jraman567 - I will refrain from hitting the send button to kick off the registration until we confirm the behaviour.  We can merge this in the meantime and raise an issue to track the testing/confirmation activity.

**Comment by jraman567** [comment]
@thomas-fossati That sounds good to me. I'll mark my comment as "Resolved" as we track the issue separately.

**Pull request merged**